File: before/data structure (1)/1966/boj1966.kt
import java.util.*

fun main() {
    val br = System.`in`.bufferedReader()

    val docs = br.readLine().toInt()

    repeat(docs) {
        val cmd = br.readLine().split(" ")
        val N = cmd[0].toInt()
        val M = cmd[1].toInt()

        val queue = LinkedList<Int>(br.readLine().split(" ").map { it.toInt()} )
        val queue_index = LinkedList<Int>((0 until N).toList())

        var cnt = 1
        while (true) {
            if (queue.max() == queue.peekFirst()) {
                if (queue_index.peekFirst() == M) {
                    println(cnt)
                    break
                }
                else {
                    queue.removeFirst()
                    queue_index.removeFirst()
                    cnt++
                }
            }
            else {
                Collections.rotate(queue, -1)
                Collections.rotate(queue_index, -1)
            }
        }
    }
}

File: before/data structure (1)/1966/boj1966.py
import sys
from collections import deque

K = int(sys.stdin.readline())

for _ in range(K):
    N, M = map(int, sys.stdin.readline().split())

    priority = 1

    queue = deque(list(map(int, sys.stdin.readline().split())))
    queue_index = deque(i for i in range(0, len(queue)))

    while True:
        if queue[0] == max(queue):
            if queue_index[0] == M:
                print(priority)
                break
            else:
                priority += 1
                queue.popleft()
                queue_index.popleft()
        else:
            queue.rotate(-1)
            queue_index.rotate(-1)
